### PR TITLE
feat: allow passing a logger function to quirrel client

### PR DESCRIPTION
**Pull request by boredland**
currently the logging is really simple and very verbose. this gives users the options to define their own logger functions for the quirrel client.

### CODE REVIEW

**Comment by boredland-ioki** [comment]
@Skn0tt I really don't understand why for node 14 the tests fail / don't find the error message :-D can you point me into a direction?

**Comment by Skn0tt** [comment]
I think that's the UI tests, which are only run on Node v14. I'll have to look into why that's failing, not sure :/

**Comment by boredland** [comment]
will add a test later today!

**Comment by boredland** [comment]
Added some tests. Would be glad if you could have a look if this goes into the right direction even @Skn0tt

File: src/next.ts [hunk 1]
```diff
@@ -45,12 +46,14 @@ export type Queue<Payload> = Omit<
 export function Queue<Payload>(
   route: string,
   handler: QuirrelJobHandler<Payload>,
-  defaultJobOptions?: DefaultJobOptions
+  defaultJobOptions?: DefaultJobOptions,
+  logger?: CustomLogger<Payload>
```
Comment (Skn0tt) [inline src/next.ts:50]: I think we should combine these two arguments into one and make it a "options" argument. I'll make some commits to change it to how I imagine the API. Not sure when I get to it, probably within the next weeks.

Comment (Skn0tt) [reply src/next.ts]: I outlined what I'm talking about in https://github.com/quirrel-dev/quirrel/pull/1039/commits/9d9ab0e82accb5a6215ce7ae293a765e73003b1a. How do you like that?

Comment (boredland) [reply src/next.ts]: I like that a lot! Gtg from my side, of course!

**Pull request merged**